cmd/virt-apiservice-register/README.md

```diff
@@ -0,0 +1,200 @@
+        - -logtostderr
+        - -v=0
+        - -api-group=myapi.io
+        - -api-version=v1alpha1
+        - -secret-name=mysecret
+        - -service-name=myservice
+        - -target-selector=app=myserver
+      containers:
+      - name: server
+        image: myimage:latest
+        imagePullPolicy: IfNotPresent
+        args:
+        - --etcd-servers=http://localhost:2379
+        - --tls-ca-file=/mypkifiles/ca.crt
+        - --tls-cert-file=/mypkifiles/tls.crt
+        - --tls-private-key-file=/mypkifiles/tls.key
+        volumeMounts:
+        - name: pkifiles
+          mountPath: "/mypkifiles"
+          readOnly: true
+```
+
+The tool doesn't automatically approve the requested certificate, it
+will just wait for the administrator (or some other tool) to approve or
+deny it. The details will be shown in the log:
+
+```
+I: Found CA
+-----BEGIN CERTIFICATE-----
+MIICyDCCAbCgAwIBAgIBADANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwprdWJl
+6UO/0vLzTdESJUz/a0RMITXICx0dt7vbPZMjOV6cS5sW4EhfA+DnhS2ZeFw=
+-----END CERTIFICATE-----
+I: Secret 'mysecret' already exists
+I: The CN for the certificate is 'myservice.default.svc.cluster.local'
+I: Additional host name for the certificate is 'myservice.default.svc.cluster.local'
+I: Additional host name for the certificate is 'myservice.default.svc'
+I: Creating CSR 'myservice.default.svc.cluster.local' for CN 'myservice.default.svc.cluster.local'
+I: The CSR has been created
+-----BEGIN CERTIFICATE REQUEST-----
+MIICyzCCAbMCAQAwLjEsMCoGA1UEAxMjbXlzZXJ2aWNlLmRlZmF1bHQuc3ZjLmNs
+35E81nwF+X7ctMUh6F29Gb2oYulCFir3W3gpeTchxmNyqv7Ebpm9aeiBeIbICzk=
+-----END CERTIFICATE REQUEST-----
+I: Waiting for approval of the CSR
+```
+
+At this point the administrator can approve the certificate with the
+following command:
+
+```shell
+$ kubctl certificate approve myservice.default.svc.cluster.local
+```
+
+The tool will detect change automatically, and will update the secret,
+create the service, and register the API service:
+
+```
+I: The CSR has been approved
+-----BEGIN CERTIFICATE-----
+MIIDbjCCAlagAwIBAgIUWIePfJahxz66w9F9qHCxXOp7AIQwDQYJKoZIhvcNAQEL
++VE2VzrNIxLNjfeooKghw72N
+-----END CERTIFICATE-----
+I: Service service 'myservice' doesn't exist
+I: Creating service 'myservice'
+I: API service service 'myapi.io' doesn't exist
+I: Creating API service 'myapi.io'
+```
+
+Once finished the tool with exit with code 0.
+
+## Permissions required
+
+In order to run successfully the tool needs permissions. They can be
+created with a role like this:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: virt-apiserver-register
+rules:
+- apiGroups:
+  - apiregistration.k8s.io
+  resources:
+  - apiservices
+  verbs:
+  - create
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+  resourceNames:
+  - extension-apiserver-authentication
+- apiGroups:
+  - ""
+  resources:
+  - services
+  verbs:
+  - create
+  - get
+- apiGroups:
+  - ""
+  resources:
```

Comment:
no action required right now.  Just wanted to point out that when we write the RBAC policies for the kubevirt api server, we'll probably need both a ClusterRole and namespace targeted Role. We'll want to limit access to things like "secrets" to resources only in a namespace KubeVirt should have access to.